File: apps/ai-analysis/router/health.py
import logging
from fastapi import APIRouter
from utils.database import check_db_health
from utils.ai_service import check_gemini_api_key

router = APIRouter()
logger = logging.getLogger(__name__)

@router.get("/health")
async def health_check():
    """
    Health check endpoint with improved diagnostics
    """
    status = {
        "database": "healthy",
        "gemini_api": "healthy", 
        "overall": "healthy",
        "details": {}
    }
    
    # Check database connection
    db_healthy, db_error = await check_db_health()
    if not db_healthy:
        status["database"] = "unhealthy"
        status["overall"] = "degraded"
        status["details"]["database_error"] = db_error
        logger.error(f"Database health check failed: {db_error}")
    
    # Check Gemini API key
    gemini_healthy, gemini_error = check_gemini_api_key()
    if not gemini_healthy:
        status["gemini_api"] = "unhealthy"
        status["overall"] = "degraded"
        status["details"]["gemini_error"] = gemini_error
        logger.error("Gemini API key not set")
    
    # Only return 503 if we can't provide any service at all
    if status["database"] == "unhealthy" and status["gemini_api"] == "unhealthy":
        status["overall"] = "unhealthy"
        return status, 503  # Service Unavailable
    
    return status 

File: apps/x-monitoring/src/router/health.ts
import { prisma } from "@believe-x/database";
import { createLogger } from "@believe-x/shared";
import express from "express";

const router = express.Router();
const logger = createLogger("health-router");

router.get("/health", async (req, res) => {
	try {
		await prisma.$queryRaw`SELECT 1`;
		res.status(200).json({ status: "ok" });
	} catch (error) {
		logger.error("Health check failed", error);
		res
			.status(500)
			.json({ status: "error", message: "Database connection failed" });
	}
});

export default router; 